core/src/main/java/org/elasticsearch/common/lucene/search/function/FiltersFunctionScoreQuery.java

```diff
@@ -293,58 +290,64 @@ public float innerScore() throws IOException {
+                        if (docSets[i].get(docId)) {
+                            maxFactor = Math.max(functions[i].score(docId, subQueryScore), maxFactor);
+                        }
                     }
-                }
-                if (maxFactor != Float.NEGATIVE_INFINITY) {
-                    factor = maxFactor;
-                }
-            } else if (scoreMode == ScoreMode.Min) {
-                double minFactor = Double.POSITIVE_INFINITY;
-                for (int i = 0; i < filterFunctions.length; i++) {
-                    if (docSets[i].get(docId)) {
-                        minFactor = Math.min(functions[i].score(docId, subQueryScore), minFactor);
+                    if (maxFactor != Float.NEGATIVE_INFINITY) {
+                        factor = maxFactor;
                     }
-                }
-                if (minFactor != Float.POSITIVE_INFINITY) {
-                    factor = minFactor;
-                }
-            } else if (scoreMode == ScoreMode.Multiply) {
-                for (int i = 0; i < filterFunctions.length; i++) {
-                    if (docSets[i].get(docId)) {
-                        factor *= functions[i].score(docId, subQueryScore);
+                    break;
+                case MIN:
+                    double minFactor = Double.POSITIVE_INFINITY;
+                    for (int i = 0; i < filterFunctions.length; i++) {
+                        if (docSets[i].get(docId)) {
+                            minFactor = Math.min(functions[i].score(docId, subQueryScore), minFactor);
+                        }
                     }
-                }
-            } else { // Avg / Total
-                double totalFactor = 0.0f;
-                float weightSum = 0;
-                for (int i = 0; i < filterFunctions.length; i++) {
-                    if (docSets[i].get(docId)) {
-                        totalFactor += functions[i].score(docId, subQueryScore);
-                        if (filterFunctions[i].function instanceof WeightFactorFunction) {
-                            weightSum+= ((WeightFactorFunction)filterFunctions[i].function).getWeight();
-                        } else {
-                            weightSum++;
+                    if (minFactor != Float.POSITIVE_INFINITY) {
+                        factor = minFactor;
+                    }
+                    break;
+                case MULTIPLY:
+                    for (int i = 0; i < filterFunctions.length; i++) {
+                        if (docSets[i].get(docId)) {
+                            factor *= functions[i].score(docId, subQueryScore);
                         }
                     }
-                }
-                if (weightSum != 0) {
-                    factor = totalFactor;
-                    if (scoreMode == ScoreMode.Avg) {
-                        factor /= weightSum;
+                    break;
+                default: // Avg / Total
```

Comment:
The comment is a bit weird here. Surely this should either be `// Avg` or `// Sum / Total`?